js/components/multi-audio/multiAudioPopover.js

```diff
@@ -0,0 +1,62 @@
+var React = require('react'),
+  CONSTANTS = require('../../constants/constants'),
+  AccessibleMenu = require('../higher-order/accessibleMenu'),
+  CloseButton = require('../closeButton');
+
+var MultiAudioPopover = React.createClass({
+  /**
+   * close the popover
+   */
+  handleClose: function() {
+    this.props.togglePopoverAction({
+      restoreToggleButtonFocus: true
+    });
+  },
+
+  /**
+   * Call setCurrentAudio from controller
+   * @param {String} id - id of the audio track to activate
+   */
+  handleMultiAudioClick: function(id) {
+    //change audio only if try to set not current audio track
+    if (this.props.controller && this.props.controller.state.currentAudioId !== id) {
+      if (typeof this.props.controller.setCurrentAudio === 'function') {
+        this.props.controller.setCurrentAudio(id);
+      }
+    }
+  },
+
+  render: function() {
+    var multiAudio = [];
+    if (this.props.controller && this.props.controller.state && !!this.props.controller.state.multiAudio) {
+      multiAudio = this.props.controller.state.multiAudio.multiAudio;
```

Comment:
Is it possible to pass this in to this component as prop? This would make this component less dependent on the controller.

Reply:
@aeng7 But we still will use functions from controller in this component. Do you really think that it is necessary? Thank you.

Reply:
Having these defined in the propTypes is fine as well. You're correct in that we depend on a lot of functions in the controller. We should still try to isolate things when possible.